scripts/generate-release-cue.rb

```diff
@@ -109,6 +110,105 @@ def create_log_file!(current_commits, new_version)
   release_log_path
 end
 
+def retire_changelog_entries!(new_version)
+  Dir.glob("#{CHANGELOG_DIR}/*.md") do |fname|
+    if File.basename(fname) == "README.md"
+      next
+    end
+    cmd = "git mv " + fname + " " + CHANGELOG_DIR + "/" + new_version.to_json + "/"
```

Comment:
BTW here's why I recommend removing them instead of moving them:
1. The contents have effectively been "moved" into the generated changelog.
2. The original contents will still exist in git in perpetuity, even if the file is removed. It just takes a little more work to pull them back out (i.e. `git checkout OLD_SHA changelog.d/FILENAME`).
3. Vector has had >11,000 commits since its inception. If most of these had a `changelog.d` entry, checking out the repo would drag out several thousand files just for the changelogs that we will effectively never look at again.

Reply:
Been discussing this a bit synchronously. If we kept them, we'd only be keeping a release or two archived so point 3) is less of a concern. I do think the "pure" thing to do is remove them.

Reply:
addressed in https://github.com/vectordotdev/vector/pull/19429/commits/a361dccc090b5daad39a80f9e5e4bf6ff61f4e54

(removing them instead of moving)